core_api/src/routes/chat.py

```diff
@@ -109,7 +106,13 @@ async def build_retrieval_chain(
 
     standalone_question = condense_question_chain({"question": question, "chat_history": previous_history})["text"]
 
-    search_kwargs = {"filter": {"term": {"creator_user_uuid.keyword": str(user_uuid)}}}
+    search_kwargs = {"filter": {"bool": {"must": [{"term": {"creator_user_uuid.keyword": str(user_uuid)}}]}}}
+
+    if chat_request.selected_files is not None:
```

Comment:
```suggestion
    if chat_request.selected_files:
```

`selected_files` should never be `None` - the model uses a default factory.